### PR TITLE
Fixed attempt to save an empty array and it saves empty string instead of null

### PR DESCRIPTION
**Pull request by dpetrovaliev**
Hey, @Gummibeer I found an issue in my own contribution I pushed a few weeks ago. I find that if you try to save an empty array it saves an empty string instead of a null.

### CODE REVIEW

File: src/Casts/EnumCollectionCast.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ public function get($model, string $key, $value, array $attributes)
      */
     public function set($model, string $key, $value, array $attributes)
     {
-        if (is_null($value)) {
+        if (is_null($value) || empty($value)) {
```
Comment (Gummibeer) [inline src/Casts/EnumCollectionCast.php:53]: Hey,
I was thinking a lot on that one and it's something specific to the `SET/COMMA` format. The default `JSON` format perfectly handles this case.
The issue of the `COMMA` format is that the `explode` call creates an array with an empty string as one value. So it's more a deserialization issue.
The check should be in the `fromDatabase()` method and return an empty array if it's an empty string.
That one here manipulates the value - an empty array is another value than `null`. And that's what the JSON format maintains, so both formats should behave the same here. Empty array in and empty array out.

**Review by Gummibeer — CHANGES_REQUESTED** [review]

File: src/Casts/EnumCollectionCast.php (hunk 1 above)
Comment (dpetrovaliev) [reply src/Casts/EnumCollectionCast.php]: @Gummibeer 

Sorry for being late with my changes. I've just pushed the requested changes.

**Review by Gummibeer — APPROVED** [review]
Thanks! 🙂

You should check your IDE settings. 😉
https://github.com/spatie/laravel-enum/pull/73/commits/29eee12d0c702ab8b14b332bbacfa41e0338c72d

**Comment by Gummibeer** [comment]
https://github.com/spatie/laravel-enum/releases/tag/2.5.1

**Pull request merged**